src/Symfony/Component/PasswordHasher/Hasher/NativePasswordHasher.php

```diff
@@ -52,15 +52,18 @@ public function __construct(int $opsLimit = null, int $memLimit = null, int $cos
         $algos = [1 => \PASSWORD_BCRYPT, '2y' => \PASSWORD_BCRYPT];
 
         if (\defined('PASSWORD_ARGON2I')) {
-            $this->algo = $algos[2] = $algos['argon2i'] = (string) \PASSWORD_ARGON2I;
+            $algos[2] = $algos['argon2i'] = (string) \PASSWORD_ARGON2I;
         }
 
         if (\defined('PASSWORD_ARGON2ID')) {
-            $this->algo = $algos[3] = $algos['argon2id'] = (string) \PASSWORD_ARGON2ID;
+            $algos[3] = $algos['argon2id'] = (string) \PASSWORD_ARGON2ID;
         }
 
         if (null !== $algo) {
             $this->algo = $algos[$algo] ?? $algo;
+        } else {
+            // use bcrypt as default
+            $this->algo = $algos[1];
         }
```

Comment:
I think this is redundant as it already defaults to bcrypt: https://github.com/symfony/symfony/blob/163df1e6731b97e3d28c7f4a8340c00f6964601d/src/Symfony/Component/PasswordHasher/Hasher/NativePasswordHasher.php#L28

I guess the changes of `$this->algo` to `$algos[...]` above are what was needed to use bcrypt as default again.

Reply:
I missed that default prop, thanks!